### PR TITLE
[MINOR]: Simplifications Sort Operator

### PR DESCRIPTION
**Pull request by akurmustafa**
## Which issue does this PR close?

Closes #.

## What changes are included in this PR?
Minor changes in the Sort code to improve readability.

### CODE REVIEW

File: datafusion/physical-plan/src/sorts/sort.rs [hunk 1]
```diff
@@ -616,11 +617,7 @@ pub fn sort_batch(
         lexsort_to_indices(&sort_columns, fetch)?
     };
 
-    let columns = batch
-        .columns()
-        .iter()
-        .map(|c| take(c.as_ref(), &indices, None))
-        .collect::<Result<_, _>>()?;
+    let columns = get_arrayref_at_indices(batch.columns(), &indices)?;
```
Comment (alamb) [inline datafusion/physical-plan/src/sorts/sort.rs:620]: 👍 

Maybe we should rename `get_arrayref_at_indices` to `take_arrays` or something 🤔  (as a follow on PR) to be more consistent

**Review by alamb — APPROVED** [review]
Thank you @akurmustafa  -- this looks like a very nice cange to me

👋  it is also great to see you back making PRs ❤️

File: datafusion/physical-plan/src/sorts/sort.rs (hunk 1 above)
Comment (akurmustafa) [reply datafusion/physical-plan/src/sorts/sort.rs:620]: This makes sense to me

**Comment by alamb** [comment]
Thanks @akurmustafa

**Pull request merged**